File: Demos/Datasnap/Server/proxy/java_android/com/embarcadero/javaandroid/TDBXDateValue.java
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

package com.embarcadero.javaandroid;

/**
 * 
 * Wraps the Date type and allows it to be null
 *
 */

public class TDBXDateValue extends DBXValue  {
	protected boolean ValueNull = false;
	private int DBXDateValue;

	public TDBXDateValue() {
		super();
		setDBXType(DBXDataTypes.DateType);
	}
	
	public boolean isNull() {
		return ValueNull;
	}
	
	@Override
	public void setNull() {
		ValueNull = true;
		DBXDateValue = 0;
	}
	
	@Override
	public void SetAsTDBXDate(int Value) throws DBXException {		
		DBXDateValue = Value;
		ValueNull = false;
	}

	public int GetAsTDBXDate() throws DBXException {
		return DBXDateValue;
	}

}

File: Demos/Datasnap/Server/proxy/objectivec_ios81/Proxy/TDBXDateValue.m
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

#import "TDBXDateValue.h"


@implementation TDBXDateValue
-(id)init{
	self = [super init];
	if (self) {
		[self setDBXType:DateType];
	}
	return self;
}
-(void) SetNull {
	ValueNull = YES;
	DBXInternalValue = 0;
}
-(bool) isNull {
	return ValueNull;
}

-(void) SetAsTDBXDate:(long) value{
	ValueNull = NO;
	DBXInternalValue = value;
}

-(long) GetAsTDBXDate {
	return DBXInternalValue;
}


@end
